### PR TITLE
docs(Toggle): Refresh toggle documentation

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Result from pairing session :) 

* Updates Toggle JSDoc and types
* Adds stories for clearer UI states

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.19-alpha.bb1657.0<br/>@codecademy/gamut-labs@8.0.3-alpha.bb1657.0<br/>@codecademy/gamut@20.0.6-alpha.bb1657.0<br/>@codecademy/styleguide@23.1.2-alpha.bb1657.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60185b2a016e9714831d76e6--gamut-preview.netlify.app>https://60185b2a016e9714831d76e6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60185b2a016e9714831d76e6>Deploy Logs</a></p>

File: packages/gamut/src/Toggle/index.tsx [hunk 1]
```diff
@@ -14,12 +14,17 @@ export type LabelProps = {
 };
 
 export type ToggleProps = {
+  /** The state of the checkbox input (this can be out of sync with the input itself if not passed) */
   checked?: boolean;
   className?: string;
-  onChange?: (...args: any[]) => any;
+  /** Called when the input is changed, but does not change the state internally you must pass `checked` to control it. */
+  onChange?: (event: React.FormEvent<HTMLInputElement>) => void;
+  /** A hidden label used for accessibility this must be unique to the page */
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Toggle/index.tsx:22]:
```suggestion
  /** A hidden label used for accessibility, unique to the page */
```

File: packages/gamut/src/Toggle/index.tsx [hunk 2]
```diff
@@ -14,12 +14,17 @@ export type LabelProps = {
 };
 
 export type ToggleProps = {
+  /** The state of the checkbox input (this can be out of sync with the input itself if not passed) */
   checked?: boolean;
   className?: string;
-  onChange?: (...args: any[]) => any;
+  /** Called when the input is changed, but does not change the state internally you must pass `checked` to control it. */
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Toggle/index.tsx:20]: Err, sorry, I don't follow what this means? Run-on sentence?

File: packages/styleguide/stories/Atoms/Toggle.stories.mdx [hunk 3]
```diff
@@ -23,34 +25,59 @@ Receives its `checked` prop from its parent. This is the default theme and size.
...
   </Story>
 </Canvas>
 
-## Size Variants
+## Size
+
+We support 2 sizes for `<Toggle />`.
+
+- `medium`: The default and preferred size, especially if it is part of a form.
+- `small`: Used for UI toggles that are outside of a form context, and may need to fit into smaller areas.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Atoms/Toggle.stories.mdx:44]:
```suggestion
- `small`: Used for UI toggles that are outside of a form context and need to fit into smaller areas.
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Some nits around phrasing, but LGTM in general! 🌮

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.19-alpha.6a85f9.0<br/>@codecademy/gamut-labs@8.0.3-alpha.6a85f9.0<br/>@codecademy/gamut@20.0.6-alpha.6a85f9.0<br/>@codecademy/styleguide@23.1.2-alpha.6a85f9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601864509c407a00c11adcb2--gamut-preview.netlify.app>https://601864509c407a00c11adcb2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601864509c407a00c11adcb2>Deploy Logs</a></p>

**Review by casey-speer — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.25-alpha.dcd80c.0<br/>@codecademy/gamut-labs@8.3.2-alpha.dcd80c.0<br/>@codecademy/gamut@21.2.2-alpha.dcd80c.0<br/>@codecademy/styleguide@24.2.4-alpha.dcd80c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601aeabe26d36916f8b8fe51--gamut-preview.netlify.app>https://601aeabe26d36916f8b8fe51--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601aeabe26d36916f8b8fe51>Deploy Logs</a></p>

**Pull request merged**